### PR TITLE
Add tool to generate metrics documentation

### PR DESCRIPTION
**Pull request by machadovilaca**
**What this PR does / why we need it**:

This PR adds a tool to automatically generate documentation with information about exposed metrics.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tools/metricsdocs/metricsdocs.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+func (m metricList) writeOut() {
+	for _, met := range m {
+		met.writeOut()
+	}
+}
+
+const filter = "kubevirt_hpp_operator_"
+
+func parseVirtMetrics(r io.Reader, metricsList *metricList) error {
```
Comment (awels) [inline tools/metricsdocs/metricsdocs.go:105]: Minor comment, this should probably be parseHppMetrics, since virt is not involved here.

Comment (machadovilaca) [reply tools/metricsdocs/metricsdocs.go]: renamed

**Review by awels — APPROVED** [review]
/lgtm
/approve

**Pull request merged**